src/frontend/src/lib/components/dapps/DappModal.svelte

```diff
@@ -0,0 +1,112 @@
+<script lang="ts">
+	import { Html, Modal } from '@dfinity/gix-components';
+	import { nonNullish } from '@dfinity/utils';
+	import DappTags from '$lib/components/dapps/DappTags.svelte';
+	import IconGitHub from '$lib/components/icons/IconGitHub.svelte';
+	import IconTwitter from '$lib/components/icons/IconTwitter.svelte';
+	import ExternalLink from '$lib/components/ui/ExternalLink.svelte';
+	import ExternalLinkIcon from '$lib/components/ui/ExternalLinkIcon.svelte';
+	import Img from '$lib/components/ui/Img.svelte';
+	import ImgBanner from '$lib/components/ui/ImgBanner.svelte';
+	import { i18n } from '$lib/stores/i18n.store';
+	import { modalStore } from '$lib/stores/modal.store';
+	import type { DappDescription } from '$lib/types/dappDescription';
+	import { replacePlaceholders } from '$lib/utils/i18n.utils';
+
+	export let dAppDescription: DappDescription;
+	$: ({ website, screenshots, twitter, github, tags, name, description, logo } = dAppDescription);
+
+	let isURLValid: boolean;
+	let formattedUrl: string | undefined;
```

Comment:
To spare two potential variables that are not in sync, you can use a type `formattedUrl: OptionString` and set the value to `null` of the parsing fails in error.

We generally use this pattern "undefined === not set, null == we set that explicitely to null" for those use cases.